contracts/src/Verification.sol

```diff
@@ -142,21 +142,24 @@ library Verification {
         bool isV2
     ) internal pure returns (bool) {
         for (uint256 i = 0; i < header.digestItems.length; i++) {
-            if (
-                header.digestItems[i].kind == DIGEST_ITEM_OTHER
-                    && header.digestItems[i].data.length == 33
-                    && header.digestItems[i].data[0] == DIGEST_ITEM_OTHER_SNOWBRIDGE
-                    && commitment == bytes32(header.digestItems[i].data[1:])
-            ) {
-                return true;
-            }
-            if (
-                isV2 && header.digestItems[i].kind == DIGEST_ITEM_OTHER
-                    && header.digestItems[i].data.length == 33
-                    && header.digestItems[i].data[0] == DIGEST_ITEM_OTHER_SNOWBRIDGE_V2
-                    && commitment == bytes32(header.digestItems[i].data[1:])
-            ) {
-                return true;
+            // First check if the digest item is of the correct kind (DIGEST_ITEM_OTHER)
```

Comment:
Potential simplification we implement to reduce the amount of code here.
```
bytes1 digestItemOtherKind =   isV2 ? DIGEST_ITEM_OTHER_SNOWBRIDGE_V2 : DIGEST_ITEM_OTHER_SNOWBRIDGE;
```

Then can check whether `header.digestItems[i].data[0] == digestItemOtherKind`

Reply:
Good catch: https://github.com/Snowfork/snowbridge/pull/1438/commits/737673a7ad82a84fa7e2ea97b13edc2717ea789a